### PR TITLE
[1.9] Backports and Documentation Fixes

### PR DESCRIPTION
**Pull request by ktbarrett**
Backports #4089 and #4095 and fixes some issues with the documentation notice by @jonpovey.

### CODE REVIEW

File: documentation/source/install.rst [hunk 1]
```diff
@@ -85,7 +85,7 @@ Installation of cocotb
 
     .. parsed-literal::
 
-        pip install cocotb =~ |version|
+        pip install "cocotb~=|version|""
```
Comment (jonpovey) [inline documentation/source/install.rst:88]: Is that a mismatched double close quote?

Comment (ktbarrett) [reply documentation/source/install.rst:88]: Yep. Damn editor.

**Review by cmarqu — APPROVED** [review]

**Comment by imphil** [comment]
Thanks for those backports. Please don't merge them yet, I'm working on some automation that I'd like to test out first.

**Comment by imphil** [comment]
Looks like our new backports GH action is intelligent enough to find the backports for #4089 in here (https://github.com/cocotb/cocotb/actions/runs/10457942861/job/28958681130?pr=4089). Cool, but that means we'll just do the manual backports here.

**Review by imphil — COMMENTED** [review]
The documentation changes need to go into `master` first and then into the stable branch. Can you remove them from this PR, @ktbarrett, or should I do that?

**Comment by ktbarrett** [comment]
I split it out. I guess this PR should close and just use the new labels?

**Comment by imphil** [comment]
I opened https://github.com/cocotb/cocotb/pull/4111 with pretty much the same content as this one here. (Officially) Not to get my PR count up, but to test out some of the backport automation and tooling. Turns out, we're not fully there yet where everything is automated: partially due to us changing the directory layout and associated merge conflicts, partially due to bugs in the backport tool (https://github.com/sorenlouv/backport/pull/505, https://github.com/sorenlouv/backport/pull/506) that got me into writing some TypeScript this evening.

**Pull request closed without merging**